#5480: Add support for EnqueueReadBuffer, EnqueueWriteBuffer, Finish and recording events

abhullar-tt opened the pull request.

tooniz committed:
#6020: Implement release trace method to undo the pinning of trace buffers to memory

abhullar-tt committed:
#5480: Add support for EnqueueReadBuffer, EnqueueWriteBuffer, Finish and recording events

abhullar-tt commented:
If enqueue api is `blocking=true` Finish is enqueued, otherwise we enqueue a "record event"

tt-rkim commented on .github/workflows/run-profiler-regression.yaml:
```diff
@@ -34,7 +34,7 @@ jobs:
-        timeout-minutes: 30
-        run: |
-          ./tests/scripts/run_profiler_regressions.sh
+      # - name: Run profiler regression tests
```
If you really don't want to run these tests, take them out of the all post commits yaml and get rid of its `push:` triggers.

Otherwise you will be taking up time on silicon runners.

tt-aho committed:
#5480: Fix prefetch pcie wrap check

pgkeller committed:
#5480: Fix packed write test

Was generating unsupported data ranges

kmabeeTT committed:
#5480: Update run_paged_tests.sh for packed-write perf tests

 - Use device iterations instead of host iterations to get higher
   BW than before. Required since test_dispatcher was updated to
   use slow-dispatch, host iterations are slow.

 - Biggest test is hanging still, under debug, so leave it alone.

kmabeeTT committed:
#5480: Add test_prefetcher.cpp tests back to CI and update test numbers

 - Also, improve some BW/iterations logging in test_dispatcher.cpp
   now that tests will use device iterations instead of host iterations
 - Add host test too (-t6)

abhullar-tt committed:
#0: Enable some enqueue rd/wr tests in CI
